### ISSUE TITLE
Nomad uniqueness of consul agent is confusing

### ISSUE THREAD

**Issue by notnoop** [hashicorp/nomad#7639]
Nomad agents are expected to have unique mapping to consul agents: No two nomad agents should communicate to the same consul agent.

Though documented[1], the failure mode is very confusing and is hard to debug.  It appears as nomad job services get removed unexpectedly and for no obvious reason, and without any log messages.

We ought to improve the situation.  Consider the following:

* Alert operators better when condition is not met, e.g
  * Nomad agent report an error or a warning on launch if two nomad agents are interacting with the same agent
  * Use consul's lock/sessions to ensure this properly

* Remove that restriction and allow multiple nomad agents to connect to the same consul
  * When removing querying consul, use labels/tags to filter by nomad agent

[1] https://github.com/hashicorp/nomad/blob/v0.11.0-beta2/website/pages/docs/configuration/consul.mdx#consul-stanza .

**Comment by shoenig** [comment]
Since this is an error condition that keeps coming up, it does seem like there must be _some_ desire to run in this configuration.

We'll probably need our queries to filter on our own generated IDs though - I suspect inserting tags is going to be unexpected for a lot of environments

**Comment by tgross** [comment]
> Since this is an error condition that keeps coming up, it does seem like there must be some desire to run in this configuration.

I'm pretty sure this configuration breaks service discovery as well, so it's not just a Nomad-specific misconfiguration.

**Comment by ultra-baba** [comment]
> > Since this is an error condition that keeps coming up, it does seem like there must be some desire to run in this configuration.
> 
> I'm pretty sure this configuration breaks Consul service discovery in the general, so it's not just a Nomad-specific misconfiguration.

By ip address?
please review my PR https://github.com/hashicorp/nomad/pull/7597

**Comment by tgross** [comment]
Closed b/c we closed out the discussion in #7819

**Comment by notnoop** [comment]
I don't think this is ought to close.  It's still confusing and better error messages are still appreciated.

In #7918, we decided that we don't want to support multiple nomad agents connecting to the same consul agent using tags - not that we no longer consider the situation confusing.
